### PR TITLE
Add an API to retrieve dates with data for ObsCollection cache

### PR DESCRIPTION
**Pull request by shifucun**

### CODE REVIEW

**Review by beets — APPROVED** [review]

File: proto/mixer.proto [hunk 1]
```diff
@@ -773,4 +773,17 @@ service Mixer {
       body: "*"
     };
   }
+
+  // Given ancestor place, place type and stat vars, return the dates that have
+  // data for each stat var.
```
Comment (pradh) [inline proto/mixer.proto:778]: ... for each stat var "across all child places" ?

File: proto/mixer.proto [hunk 2]
```diff
@@ -773,4 +773,17 @@ service Mixer {
       body: "*"
     };
   }
+
+  // Given ancestor place, place type and stat vars, return the dates that have
```
Comment (pradh) [inline proto/mixer.proto:777]: child place type ?

File: proto/stat.proto [hunk 3]
```diff
@@ -274,4 +277,16 @@ message GetStatSetRequest {
 message GetStatSetResponse {
   // Keyed by statVar.
   map<string, PlacePointStat> data = 1;
+}
+
+
+message GetPlaceStatDateWithinPlaceRequest {
```
Comment (pradh) [inline proto/stat.proto:283]: nit: I know it makes it long, but I wonder if we should qualify it as WithinAncestorPlace ?

File: internal/server/place_stat_date.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+	for token, data := range cacheData {
+		if data != nil {
+			cohorts := data.(*pb.ObsCollection).SourceCohorts
+			sort.Sort(SeriesByRank(cohorts))
+			dates := []string{}
+			for date := range cohorts[0].Val {
+				dates = append(dates, date)
+			}
+			sort.Strings(dates)
+			result.Data[token] = &pb.DateList{Dates: dates}
```
Comment (pradh) [inline internal/server/place_stat_date.go:93]: nit: maybe a TODO to refactor this part into a helper?

**Review by pradh — APPROVED** [review]

File: proto/stat.proto (hunk 3 above)
Comment (shifucun) [reply proto/stat.proto:283]: here aligns with other "within-place" api names: https://docs.google.com/document/d/12rdNsaLsMj5LxUWS3dynC8fuYTd0sfxqHio0Hh8Joao/edit#heading=h.9ssx4uezxkkc

what do you think?

File: proto/mixer.proto (hunk 2 above)
Comment (shifucun) [reply proto/mixer.proto:777]: done

File: proto/stat.proto (hunk 3 above)
Comment (pradh) [reply proto/stat.proto:283]: Alright then...

**Pull request merged**